indexing-service/src/main/java/io/druid/indexing/overlord/autoscaling/TwoCloudWorkerProvisioningStrategy.java

```diff
@@ -0,0 +1,161 @@
+  {
+    @Override
+    public ScheduledExecutorService get()
+    {
+      return ScheduledExecutors.fixed(1, "TwoCloudWorkerProvisioningStrategy-manager--%d");
+    }
+  };
+
+  private final String taskLabel1;
+  private final String ipPrefix1;
+  private final PendingTaskBasedWorkerProvisioningStrategy provisioningStrategy1;
+  private final String taskLabel2;
+  private final String ipPrefix2;
+  private final PendingTaskBasedWorkerProvisioningStrategy provisioningStrategy2;
+
+  @Inject
+  public TwoCloudWorkerProvisioningStrategy(
+      Supplier<TwoCloudConfig> twoCloudConfig,
+      PendingTaskBasedWorkerProvisioningConfig config,
+      ProvisioningSchedulerConfig provisioningSchedulerConfig
+  )
+  {
+    this(
+        twoCloudConfig.get().getTaskLabel1(),
+        twoCloudConfig.get().getIpPrefix1(),
+        twoCloudConfig.get().getWorkerBehaviorConfig1(),
+        twoCloudConfig.get().getTaskLabel2(),
+        twoCloudConfig.get().getIpPrefix2(),
+        twoCloudConfig.get().getWorkerBehaviorConfig2(),
+        config,
+        provisioningSchedulerConfig
+    );
+  }
+
+  public TwoCloudWorkerProvisioningStrategy(
+      String taskLabel1,
+      String ipPrefix1,
+      WorkerBehaviorConfig workerBehaviorConfig1,
+      String taskLabel2,
+      String ipPrefix2,
+      WorkerBehaviorConfig workerBehaviorConfig2,
+      PendingTaskBasedWorkerProvisioningConfig config,
+      ProvisioningSchedulerConfig provisioningSchedulerConfig
+  )
+  {
+    super(provisioningSchedulerConfig, EXEC_FACTORY);
+    this.taskLabel1 = taskLabel1;
+    this.taskLabel2 = taskLabel2;
+    this.ipPrefix1 = ipPrefix1;
+    this.ipPrefix2 = ipPrefix2;
+    provisioningStrategy1 = new PendingTaskBasedWorkerProvisioningStrategy(
+        config,
+        Suppliers.ofInstance(workerBehaviorConfig1),
+        provisioningSchedulerConfig,
+        EXEC_FACTORY,
+        ipPrefix1
+    );
+    provisioningStrategy2 = new PendingTaskBasedWorkerProvisioningStrategy(
+        config,
+        Suppliers.ofInstance(workerBehaviorConfig2),
+        provisioningSchedulerConfig,
+        EXEC_FACTORY,
+        ipPrefix2
+    );
+  }
+
+  @Override
+  Provisioner makeProvisioner(TasksAndWorkers runner)
+  {
+    final TaskPredicate taskPredicate1 = new TaskPredicate(taskLabel1, true);
+    final TaskPredicate taskPredicate2 = new TaskPredicate(taskLabel2, false);
+
+    final Provisioner provisioner1 = provisioningStrategy1.makeProvisioner(
+        new TasksAndWorkersFilteredByIp((WorkerTaskRunner) runner, ipPrefix1, taskPredicate1)
+    );
+    final Provisioner provisioner2 = provisioningStrategy2.makeProvisioner(
+        new TasksAndWorkersFilteredByIp((WorkerTaskRunner) runner, ipPrefix2, taskPredicate2)
+    );
+    return new Provisioner()
+    {
+      @Override
+      public boolean doTerminate()
+      {
+        return provisioner1.doTerminate() || provisioner2.doTerminate();
```

Comment:
the second one won't fire if the first one returns true, if I recall. This should fire both regardless of either's return value